lib/main.dart

```diff
@@ -108,35 +116,54 @@ class _ParentWidgetState extends ConsumerState<ParentWidget>
     );
     onMessageAppOpened(ref);
     initializeNotification(ref);
-    // listened for app background/foreground events
     WidgetsBinding.instance.addObserver(this);
   }
 
   @override
   Widget build(BuildContext context) {
     final goRouter = ref.watch(goRouterProvider);
-    ref.listen(deviceAndAppInfoProvider, (_, info) {
-      if (info.hasValue) {
-        var val = info.value;
-        var appOpenedModel = AppOpenedModel(
-          deviceOs: val?.os ?? '',
-          deviceLanguage: val?.languageCode ?? '',
-          deviceModel: val?.model ?? '',
-          buildNumber: val?.buildNumber ?? '',
-          appVersion: val?.appVersion ?? '',
-        );
-        var event = EventsModel(
-          name: EventTypes.appOpened,
-          payload: appOpenedModel.toJson(),
-        );
-        ref.read(eventsProvider(event: event.toJson()));
-      }
-    });
 
     return MaterialApp.router(
       routerConfig: goRouter,
       theme: appTheme(context),
       title: ParentWidget._title,
     );
   }
+  
+  void _handleAudioCompletion(
+    WidgetRef ref,
```

Comment:
I don't think this should be in main either. Just pass the extras. That will make it easier to test